framework/src/systems/SystemBase.C

```diff
@@ -1328,6 +1333,9 @@ SystemBase::copyOldSolutions()
 void
 SystemBase::restoreSolutions()
 {
+  if (!hasSolutionState(1))
```

Comment:
I feel like `solution state` should be `enum` instead of `unsigned int` to increase readability.

`enum  SolutionState { current , old, older, ...,  };`

Reply:
I thought about this too. The problem is that we also use this for state 3, and it could potentially be used for even older states.

In that case - do we have:

- 0: current
- 1: old
- 2: older
- 3: oldoldold/olderer
- 4: oldoldoldold/oldererer

??

Reply:
`unsigned int` seems fine to me given that it can go arbitrarily high